### PR TITLE
Rewrite some decomps to allow producing aten ops

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #93099
* #93130

This introduces a new stop to the decomposition train.
Before reaching prims.view_of, it will stop at aten.alias. Export path wants to get off the train at aten ops.

### CODE REVIEW

File: torch/_prims_common/wrappers.py
Comment (ngimel) [inline torch/_prims_common/wrappers.py:42]: This could increase tracing time, as `to` would go through a series of redispatches, cc @chillee, @voznesenskym

Comment (Chillee) [reply torch/_prims_common/wrappers.py:42]: What's the motivation behind this PR?

Comment (ngimel) [reply torch/_prims_common/wrappers.py:42]: Backends that want only `aten` ops in the graph they are getting.

Comment (SherlockNoMad) [reply torch/_prims_common/wrappers.py:41]: This introduces a new stop to the decomposition train. 
Before reaching prims.view_of, it will stop at aten.alias. Export path want to get off train at aten ops.

**Comment by zhxchen17** [comment]
just for reference and for record, we got some failures internally, diff number: D42764068

**Comment by SherlockNoMad** [comment]
> just for reference and for record, we got some failures internally, diff number: D42764068

Noted, I will hold off landing https://github.com/pytorch/pytorch/pull/93133 until failures are fixed.

File: torch/_prims_common/wrappers.py
Comment (Chillee) [reply torch/_prims_common/wrappers.py:41]: @SherlockNoMad imo, we should just turn `prims.convert_element_type` into an ATen op. Note that `aten.to.dtype` is  *not* a composite explicit op - it decomposes into `aten._to_copy`.

Comment (SherlockNoMad) [reply torch/_prims_common/wrappers.py:41]: In this PR, I will directly use aten._to_copy instead of aten._to_copy. 
Agree that prims.convert_element_type should just be something like aten.cast.

Comment (ngimel) [reply torch/_prims_common/wrappers.py]: This introduces `aten` op to `_refs` decompositions that are meant to operate on `torch.*` API. Conceptually `.to` was better, but it introduced yet another trip through dispatched ops. Probably land it with `.to` and let's see if it regresses tracing time.

Comment (SherlockNoMad) [reply torch/_prims_common/wrappers.py:39]: OK, I reverted back to use .to()

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
Stamped to unblock, but are you sure this will actually help them? `to` will go to `_to_copy`, and if the use `_to_copy` decomposition it'll end up at the same `prims.convert_element_dtype` op. Are they stopping decomp at `_to_copy`?

**Comment by SherlockNoMad** [comment]
> Stamped to unblock, but are you sure this will actually help them? `to` will go to `_to_copy`, and if the use `_to_copy` decomposition it'll end up at the same `prims.convert_element_dtype` op. Are they stopping decomp at `_to_copy`? Changing element dtype (and just element dtype) is an op that conceptually should be prim, that's why @Chillee suggested adding it to ATen, and all backends should support it.

I have a following PR to stop decomposing at aten._to_copy. https://github.com/pytorch/pytorch/pull/93131

**Pull request closed without merging**
